handler/src/main/java/io/netty/handler/ssl/OpenSslEngine.java

```diff
@@ -60,8 +60,10 @@
 /**
  * Implements a {@link SSLEngine} using
  * <a href="https://www.openssl.org/docs/crypto/BIO_s_bio.html#EXAMPLE">OpenSSL BIO abstractions</a>.
+ *
+ * <strong>Extending this class is strightly forbidden from external projects.</strong>
  */
-public final class OpenSslEngine extends SSLEngine {
+public class OpenSslEngine extends SSLEngine {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) The Cyclomatic Complexity of this class is 418 which is greater than 300 authorized. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://garage.netty.io/sonarqube/coding_rules#rule_key=squid%3AClassCyclomaticComplexity)